docs/overview/deep_dive/ivy_lint.rst

```diff
@@ -0,0 +1,58 @@
+Ivy-Lint: Ivy's Custom Code Formatters
+======================================
+
+Overview
+--------
+
+`ivy-lint` is a specialized suite of formatters crafted for the Ivy codebase. It addresses unique formatting requirements not catered to by standard Python formatters. While the suite currently highlights the `FunctionOrderingFormatter`, we're continually expanding to include more formatters tailored to Ivy's needs.
+
+Existing Formatters
+-------------------
+
+FunctionOrderingFormatter
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This formatter ensures a standardized order of declarations within Python files, organizing functions, classes, and assignments based on a hierarchy designed for the Ivy codebase.
+
+**Purpose**: To bring a sense of uniformity and structure to the code files by sorting various Python declarations.
+
+**Target Files**: It is crafted to work on particular files that align with predefined patterns.
```

Comment:
That is a bit ominous, I guess it would be better to just say "frontends and tests" maybe?